eclipse-scout-core/src/table/columns/LookupColumn.ts

```diff
@@ -0,0 +1,172 @@
+      parent: this.table,
+      lookupCall: this.lookupCall,
+      browseHierarchy: this.browseHierarchy,
+      browseMaxRowCount: this.browseMaxRowCount
+    });
+
+    editor.on('lookupCallDone', e => {
+      if (!this.distinct) {
+        return;
+      }
+
+      const valuesInOtherRows = [];
+      this.table.rows.forEach(r => {
+        if (r === row) {
+          return;
+        }
+        valuesInOtherRows.push(...arrays.ensure(this.cellValue(r)));
+      });
+      arrays.removeAll(valuesInOtherRows, this.cellValue(row));
+      e.result.lookupRows.forEach(lookupRow => {
+        if (scout.isOneOf(lookupRow.key, valuesInOtherRows)) {
+          lookupRow.enabled = false;
+        }
+      });
+    });
+
+    // pass editor events to column
+    editor.on('prepareLookupCall', e => this.trigger('prepareLookupCall', {
+      lookupCall: e.lookupCall,
+      row
+    }));
+    editor.on('lookupCallDone', e => this.trigger('lookupCallDone', {
+      result: e.result
+    }));
+
+    return editor;
+  }
+
+  override isContentValid(row: TableRow): { valid: boolean; validByMandatory: boolean; errorStatus: Status } {
+    const validationResult = super.isContentValid(row);
+
+    if (!this.distinct) {
+      return validationResult;
+    }
+
+    const cell = this.cell(row);
+    if (!this.table.rows.find(r => r !== row && arrays.containsAny(cell.value, this.cellValue(r)))) {
+      return validationResult;
+    }
+
+    const distinctErrorStatus = Status.error(this.session.text('ui.LookupColumnDistinctErrorMessage'));
+    const errorStatus = validationResult.errorStatus ? Status.error({children: [distinctErrorStatus, validationResult.errorStatus]}) : distinctErrorStatus;
+    return {
+      ...validationResult,
+      valid: errorStatus.isValid(),
+      errorStatus
+    };
+  }
+
+  protected override _formatValue(value: TValue[], row?: TableRow): string | JQuery.Promise<string> {
+    return LookupColumn.formatValues(value, this.lookupCall, lookupCall => this.trigger('prepareLookupCall', {lookupCall, row}));
+  }
+
+  static formatValues<TValue>(values: TValue[], lookupCall: LookupCall<TValue>, prepareLookupCall?: (lookupCall: LookupCall<TValue>) => void): string | JQuery.Promise<string> {
+    if (!values?.length || !lookupCall) {
+      return '';
+    }
+    lookupCall = lookupCall.clone();
+    prepareLookupCall?.(lookupCall);
+    return lookupCall.textsByKeys(values)
+      .then(result => strings.join(', ', ...Object.values(result)));
+  }
+}
+
+export interface LookupColumnModel<TValue> extends ColumnModel<TValue[]> {
+  lookupCall?: LookupCallOrModel<TValue>;
+  /**
+   * Default is false.
+   */
+  browseHierarchy?: boolean;
+  /**
+   * A positive number, _not_ null or undefined!
+   *
+   * Default is SmartField.DEFAULT_BROWSE_MAX_COUNT.
+   */
+  browseMaxRowCount?: number;
+  /**
+   * Whether the same value can be selected in multiple rows or not.
+   *
+   * Default is `false`.
+   */
+  distinct?: boolean;
```

Comment:
Better create a separate file for Model and EventMap so it is consistent between all scout columns